src/controllers/howDoYouWantToPay.controller.ts

```diff
@@ -0,0 +1,131 @@
+import DissolutionGetResponse from 'app/models/dto/dissolutionGetResponse'
+import PaymentType from 'app/models/dto/paymentType.enum'
+import HowDoYouWantToPayFormModel from 'app/models/form/howDoYouWantToPay.model'
+import Optional from 'app/models/optional'
+import DissolutionSession from 'app/models/session/dissolutionSession.model'
+import ValidationErrors from 'app/models/view/validationErrors.model'
+import { HOW_DO_YOU_WANT_TO_PAY_URI, PAY_BY_ACCOUNT_URI, SEARCH_COMPANY_URI } from 'app/paths'
+import { howDoYouWantToPaySchema } from 'app/schemas/howDoYouWantToPay.schema'
+import DissolutionService from 'app/services/dissolution/dissolution.service'
+import PaymentService from 'app/services/payment/payment.service'
+import SessionService from 'app/services/session/session.service'
+import TYPES from 'app/types'
+import FormValidator from 'app/utils/formValidator.util'
+
+interface ViewModel {
+  data?: Optional<HowDoYouWantToPayFormModel>
+  errors?: Optional<ValidationErrors>
+}
+
+@controller(HOW_DO_YOU_WANT_TO_PAY_URI)
+export class HowDoYouWantToPayController extends BaseController {
+
+  public constructor(
+    @inject(SessionService) private session: SessionService,
+    @inject(FormValidator) private validator: FormValidator,
+    @inject(DissolutionService) private service: DissolutionService,
+    @inject(PaymentService) private paymentService: PaymentService,
+    @inject(TYPES.PAY_BY_ACCOUNT_FEATURE_ENABLED) private PAY_BY_ACCOUNT_FEATURE_ENABLED: number
+  ) {
+    super()
+  }
+
+  @httpGet('')
+  public async get(): Promise<string | RedirectResult> {
+    if (!this.PAY_BY_ACCOUNT_FEATURE_ENABLED) {
+      return Promise.reject(new NotFoundError('Feature toggle not enabled'))
+    }
+
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    if (await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    return this.renderView(dissolutionSession.howDoYouWantToPayForm)
+  }
+
+  @httpPost('')
+  public async post(@requestBody() body: HowDoYouWantToPayFormModel): Promise<string | RedirectResult>  {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    if (await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    const errors: Optional<ValidationErrors> = this.validator.validate(body, howDoYouWantToPaySchema)
+    if (errors) {
+      return this.renderView(body, errors)
+    }
+
+    const paymentType: PaymentType = body.paymentType!
+
+    this.updateSession(dissolutionSession, paymentType, body)
+
+    return this.redirect(await this.getRedirectURI(paymentType, dissolutionSession))
```

Comment:
hmmm you are calling updateSession before you have set the paymentStateUUID. Does this work?